kratos/utilities/nurbs_utilities/snake_sbm_utilities.cpp

```diff
@@ -0,0 +1,838 @@
+                                                    Vector& knotVectorU, 
+                                                    Vector& knotVectorV,
+                                                    const Parameters mParameters) { 
+        
+        // Check
+        KRATOS_ERROR_IF_NOT(mParameters.Has("sbm_parameters")) << "sbm_parameters has not been defined in the nurbs modeler" << std::endl;
+        
+        // Initilize the property of skin_model_part_in and out
+        if (rSkinModelPartInnerInitial.Nodes().size()>0) {
+        
+            CreateTheSnakeCoordinates(rIgaModelPart, rSkinModelPartInnerInitial, rSkinModelPart, rEchoLevel, knotVectorU, knotVectorV, mParameters, true);
+                
+        }
+        if (rSkinModelPartOuterInitial.Nodes().size()>0) {
+
+            CreateTheSnakeCoordinates(rIgaModelPart, rSkinModelPartOuterInitial, rSkinModelPart, rEchoLevel, knotVectorU, knotVectorV, mParameters, false);
+        }
+    }   
+
+    void SnakeSbmUtilities::CreateTheSnakeCoordinates(ModelPart& rIgaModelPart, 
+                                                      ModelPart& rSkinModelPartInitial, 
+                                                      ModelPart& rSkinModelPart,
+                                                      int rEchoLevel, 
+                                                      Vector& knotVectorU, 
+                                                      Vector& knotVectorV,
+                                                      const Parameters mParameters,
+                                                      bool isInner) 
+    { 
+        
+        std::string surrogate_sub_model_part_name; 
+        std::string skin_sub_model_part_name; 
+        // ModelPart skin_sub_model_part; 
+        if (isInner) {
+            surrogate_sub_model_part_name = "surrogate_inner";
+            skin_sub_model_part_name = "inner";
+        }
+        else {
+            surrogate_sub_model_part_name = "surrogate_outer";
+            skin_sub_model_part_name = "outer";
+        }
+        
+        ModelPart& skin_sub_model_part = rSkinModelPart.GetSubModelPart(skin_sub_model_part_name);
+        ModelPart& surrogate_sub_model_part = rIgaModelPart.GetSubModelPart(surrogate_sub_model_part_name);
+
+        if (!rSkinModelPartInitial.HasProperties(0)) rSkinModelPartInitial.CreateNewProperties(0);
+        if (!rSkinModelPart.HasProperties(0)) rSkinModelPart.CreateNewProperties(0);
+        Properties::Pointer p_cond_prop_in = rSkinModelPartInitial.pGetProperties(0);
+        rSkinModelPartInitial.AddProperties(p_cond_prop_in);
+
+        array_1d<double, 2> knot_step_uv(2);
+        knot_step_uv[0] = std::abs(knotVectorU[int(knotVectorU.size()/2) +1]  - knotVectorU[int(knotVectorU.size()/2)] ) ;
+        knot_step_uv[1] = std::abs(knotVectorV[int(knotVectorV.size()/2) +1]  - knotVectorV[int(knotVectorV.size()/2)] ) ;
+
+        Vector meshSizes_uv(2);
```

Comment:
```suggestion
        Vector mesh_sizes_uv(2);
```
Please use snake_case for local variables.